app/controllers/teachers_controller.rb

```diff
@@ -95,7 +101,18 @@ def edit
   def update
     load_school
     ordered_schools
-    @teacher.assign_attributes(teacher_params)
+
+    personal_emails_params = params[:teacher].keys.select { |key| key.to_s.start_with?("personal_email") }
```

Comment:
I think it looks like we should probably separate the form email fields. The parameters logic is pretty complicated here.